### PR TITLE
Support u-blox 10 series hwVersion

### PR DESCRIPTION
**Pull request by MATEKSYS**

### CODE REVIEW

File: src/main/io/gps_ublox.c [hunk 1]
```diff
@@ -823,9 +823,13 @@ STATIC_PROTOTHREAD(gpsConfigure)
...
-    // u-Blox 9
-    // M9N does not support some of the UBX 6/7/8 messages, so we have to configure it using special sequence
-    if (gpsState.hwVersion >= 190000) {
+    // M9N & M10 does not support some of the UBX 6/7/8 messages, so we have to configure it using special sequence
+    // ublox_6   hwVersion 00040007
+    // ublox_7   hwVersion 00070000
+    // ublox_M8  hwVersion 00080000
+    // ublox_M9  hwVersion 00190000
+    // ublox_M10 hwVersion 000A0000
+    if (gpsState.hwVersion > 80000) {
```
Comment (stronnag) [inline src/main/io/gps_ublox.c:832]: Shouldn't this be a hex comparison `0x80000` (and it was just good fortune that the M9N check worked).?
Or do we have a bigger issue here that we assume a base10 conversion when (as for M10) the value is actually hexadecimal and we need a more thought out solution.

**Comment by stronnag** [comment]
This needs to be properly resolved, including the legacy mis-conversion that worked by chance.

**Comment by stronnag** [comment]
If #8057 is merged, then this can fixed (until Ublox invent another versioning scheme) by:
* replacing `fastA2I` by `ascii2hex`
* comparing against hex constants `if (gpsState.hwVersion > 0x80000)`

File: src/main/io/gps_ublox.c (hunk 1 above)
Comment (digitalentity) [reply src/main/io/gps_ublox.c:832]: It appears that the value is actually hexadecimal and it's a mere chance that it worked before

Comment (digitalentity) [reply src/main/io/gps_ublox.c:832]: Actually it's even worse than that. There are no guarantees that the hw version is a number, it's a string and I think we should better be treating it as such.

**Comment by DzikuVx** [comment]
replaced by #8058

**Pull request closed without merging**